src/components/common/DeleteItemModal/DeleteItemModal.tsx

```diff
@@ -0,0 +1,128 @@
+import styles from "./DeleteItemModal.module.scss";
+import React from "react";
+import cn from "classnames";
+import { AppContext, AppContextState } from "context/AppContext";
+import { disableBodyScroll, enableBodyScroll } from "body-scroll-lock";
+import Loader from "components/common/Loader/Loader";
+import { matchPath, RouteComponentProps, withRouter } from "react-router-dom";
+import { get } from "lodash-es";
+import { COLLECTION_URL, COLLECTIONS_URL } from "../../../constants/urls";
+import { deleteItem } from "../../../context/CollectionsAPI/api";
+
+type State = {
+  isLoading: boolean;
+  itemId: number;
+};
+type Props = RouteComponentProps;
+
+class DeleteItemModal extends React.Component<Props, State> {
+  static contextType = AppContext;
+  context!: AppContextState;
+  modalTarget = null;
+
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      isLoading: false,
+      itemId: null,
+    };
+  }
+
+  onBackgroundClicked = () => {
+    this.closeModal();
+  };
+
+  closeModal = () => {
+    this.context.closeModal();
+  };
+
+  componentDidMount() {
+    this.modalTarget = document.querySelector("#deleteItemId");
+    this.disableWindowsScroll();
+    this.context.getItemId().then((itemId: any) => {
```

Comment:
You can replace all this logic for storing the itemId on the state with a simple
```
const modalParams = this.context.getModalParams()
const itemId = modalParams.collectionItemId
```
in the submit function.